File: work/socket0.c
#include <unistd.h>
#include <stdio.h>
#include <signal.h>
#include <sys/types.h>

int main()
{
    pid_t pid[8];

    for (int i = 0; i < 8; ++i)
    {
        pid[i] = fork();
        if (pid[i] == 0)
        {
            // Child.
            int e = execl(
                "/usr/bin/taskset",
                "taskset", "-c", i,
                "/home/pherna06/venv-esfinge/server-consumption/work/test");

            if (e == -1)
            {
                printf("execl error");
                return;
            }
            break;
        }        
    }

    printf("Press any key to kill processes.");
    int c = getchar();

    for (int i = 0; i < 8; ++i)
    {
        kill(pid[i], SIGKILL);
    }

    return 0;
}

File: work/socket0.py
import os
import signal

def process(core):
    command = "taskset -c " + str(core) + " "
    command += "/home/pherna06/venv-esfinge/server-consumption/work/test"

    os.system(command)

pid = []
for i in range(8):
    pid.append(os.fork())
    if pid[i] == 0: #child
        process(i)
        exit()

os.system('read -n 1 -s -r -p "Press any key to continue"')

for i in range(8):
    os.kill(pid[i], signal.SIGKILL)
